File: Inventario/DataAccess/Models/Commission.cs
using System;
using System.Collections.Generic;
using System.Text;
using Util.Enums;

namespace Data
{
    public class Commission: Entity
    {
        public Guid VendorId { get; set; }
        public Vendor Vendor { get; set; }
        public Guid? ClientId { get; set; }
        public Client Client { get; set; }
        public string ClientName { get; set; }
        public string  Product { get; set; }
        public decimal Price { get; set; }
        public ePaymentTypes PaymentType { get; set; }
        public DateTime Date { get; set; }
        public decimal Value { get; set; }
    }
}

File: inventario_frontend/src/models/commission.ts
import { Vendor } from "./vendor";
import { Client } from "./client";
import { paymentTypes } from "@/enums/paymentTypes";

export class CommissionForCreation {
  public vendorId: string | undefined = undefined;
  public vendor: Vendor | undefined = undefined;
  public clientId?: string;
  public client?: Client;
  public clientName: string | undefined = undefined;
  public product: string | undefined = undefined;
  public price: number | undefined = undefined;
  public paymentType: paymentTypes | undefined = undefined;
  public date: Date | undefined = undefined;
  public value: number | undefined = undefined;
}

export class Commission extends CommissionForCreation {
  public id!: string;
}
